app/models/edition.rb

```diff
@@ -604,19 +601,17 @@ def body_required?
   end
 
   def has_associated_needs?
-    associated_needs.any?
+    respond_to?(:need_ids) && need_ids.try(:any?)
   end
 
   def associated_needs
```

Comment:
> We weren't checking if the object responds_to?(:need_ids) before. Is that necessary?

If you don't have that check, and you call that `has_user_needs?` on some Edition subclass that is not a DetailedGuide, then you would get an error.

One alternative that I see would be to move the needs related code out of the DetailedGuide class and up to the Edition? That would then mean that need_ids would be defined for all Editions.